pgctl/fuser.py

```diff
@@ -56,16 +60,28 @@ def fuser(path):
                 yield pid
                 break
 
+            if allow_deleted and found.st_nlink == 0:
+                from os import readlink
+                if readlink(fd) == path + ' (deleted)':
```

Comment:
Pretty sure this fails if fd isn't a symlink, but that might be fine if everything in /proc/*/fd/ is already a symlink.